src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -758,15 +761,20 @@ protected function getRawObject(string $pdfData, int $offset = 0): array
                     $offset += 6;
                     if (1 == preg_match('/^([\r]?[\n])/isU', substr($pdfData, $offset, 4), $matches)) {
                         $offset += \strlen($matches[0]);
+
+                        $streamLen = (int) $this->getHeaderValue($headerDic, 'Length', 'numeric', 0);
```

Comment:
Please add one or more sentences about this new code.

Reply:
If you look through PDFs, you'll find that stream blocks (stream-endstream) are anticipated with header section (<< >>) with a field Length, that contains length of this stream.

This fact is already used in protected function decodeStream
https://github.com/smalot/pdfparser/blob/65bdad55246ec230a055fbda501c6fed592c1942/src/Smalot/PdfParser/RawData/RawDataParser.php#L100

So we can extract this length, add it to preg_match's offset and:
-- help preg_match to test less data, i.e. work faster
-- prevent (not very probable) false match, when compressed data contains 'endstream ' by itself